app/reports/responses_report.rb

```diff
@@ -0,0 +1,47 @@
+class ResponsesReport < AdminReport
+  include Chartkick::Helper
+
+  def initialize(team, time_periods)
+    super(team)
+    @time_periods = time_periods
+  end
+
+  def generate
+    response_counts = receive_response_counts
+    data = response_data(response_counts)
+
+    chart_id = SecureRandom.uuid
+    chart = create_chart(data, chart_id)
+
+    { chart: chart, id: chart_id }
+  end
+
+  private
+
+  def receive_response_counts
+    Response.joins(user: { teams: :users_teams })
+            .where(users_teams: { team_id: @team.id })
+            .where(time_period_id: @time_periods)
+            .distinct
+            .group('responses.time_period_id')
+            .count
+  end
```

Comment:
Nicely done! 👍🏻 
If this might be called more than once, you can cache the method results using [memoization](https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/):
```ruby
def receive_response_counts
   @receive_response_counts ||= begin
      ...
   end
end
```
Also please use the memoization approach for other methods that can be called multiple times.